worker-sys/src/types/utils/typed_array.rs

```diff
@@ -0,0 +1,182 @@
+
+            #[wasm_bindgen(method, getter, structural)]
+            pub fn length(this: &$name) -> u32;
+
+            #[wasm_bindgen(method, setter)]
+            pub fn set_length(this: &$name, value: u32);
+
+            #[wasm_bindgen(method)]
+            pub fn map(
+                this: &$name,
+                predicate: &mut dyn FnMut($type, u32, $name) -> $type,
+            ) -> $name;
+
+            #[wasm_bindgen(static_method_of = $name, js_name = of)]
+            pub fn of1(a: &$type) -> $name;
+
+            #[wasm_bindgen(static_method_of = $name, js_name = of)]
+            pub fn of2(a: &$type, b: &$type) -> $name;
+
+            #[wasm_bindgen(static_method_of = $name, js_name = of)]
+            pub fn of3(a: &$type, b: &$type, c: &$type) -> $name;
+
+            #[wasm_bindgen(static_method_of = $name, js_name = of)]
+            pub fn of4(a: &$type, b: &$type, c: &$type, d: &$type) -> $name;
+
+            #[wasm_bindgen(static_method_of = $name, js_name = of)]
+            pub fn of5(a: &$type, b: &$type, c: &$type, d: &$type, e: &$type) -> $name;
+
+            #[wasm_bindgen(method)]
+            pub fn pop(this: &$name) -> $type;
+
+            #[wasm_bindgen(method)]
+            pub fn push(this: &$name, value: &$type) -> u32;
+
+            #[wasm_bindgen(method)]
+            pub fn reduce(
+                this: &$name,
+                predicate: &mut dyn FnMut($type, $type, u32, $name) -> $type,
+                initial_value: &$type,
+            ) -> $type;
+
+            #[wasm_bindgen(method, js_name = reduceRight)]
+            pub fn reduce_right(
+                this: &$name,
+                predicate: &mut dyn FnMut($type, $type, u32, $name) -> $type,
+                initial_value: &$type,
+            ) -> $type;
+
+            #[wasm_bindgen(method)]
+            pub fn reverse(this: &$name) -> $name;
+
+            #[wasm_bindgen(method)]
+            pub fn shift(this: &$name) -> $type;
+
+            #[wasm_bindgen(method)]
+            pub fn slice(this: &$name, start: u32, end: u32) -> $name;
+
+            #[wasm_bindgen(method)]
+            pub fn some(this: &$name, predicate: &mut dyn FnMut($type) -> bool) -> bool;
+
+            #[wasm_bindgen(method)]
+            pub fn sort(this: &$name) -> $name;
+
+            #[wasm_bindgen(method)]
+            pub fn splice(this: &$name, start: u32, delete_count: u32, item: &$type) -> $name;
+
+            #[wasm_bindgen(method, js_name = toLocaleString)]
+            pub fn to_locale_string(
+                this: &$name,
+                locales: &$type,
+                options: &$type,
+            ) -> ::js_sys::JsString;
+
+            #[wasm_bindgen(method, js_name = toString)]
+            pub fn to_string(this: &$name) -> ::js_sys::JsString;
+
+            #[wasm_bindgen(method)]
+            pub fn unshift(this: &$name, value: &$type) -> u32;
+        }
+    };
+}
```

Comment:
How is this file different to what is already in web_sys, and can't we just use that directly?